PhysicsTools/PatAlgos/plugins/PATPackedCandidateProducer.cc

```diff
@@ -314,13 +314,23 @@ void pat::PATPackedCandidateProducer::produce(edm::StreamID, edm::Event& iEvent,
 
 	if(abs(cand.pdgId()) == 1 || abs(cand.pdgId()) == 130) {
 	  outPtrP->back().setHcalFraction(cand.hcalEnergy()/(cand.ecalEnergy()+cand.hcalEnergy()));
-        } else if(isIsolatedChargedHadron) {
-          outPtrP->back().setRawCaloFraction((cand.rawEcalEnergy()+cand.rawHcalEnergy())/cand.energy());
-          outPtrP->back().setHcalFraction(cand.rawHcalEnergy()/(cand.rawEcalEnergy()+cand.rawHcalEnergy()));
-	} else {
+        } else if (cand.charge() && cand.pt() > 0.5) {
+          outPtrP->back().setHcalFraction(cand.hcalEnergy()/(cand.ecalEnergy()+cand.hcalEnergy()));
+          outPtrP->back().setCaloFraction((cand.hcalEnergy() + cand.ecalEnergy())/cand.energy());
```

Comment:
Based on 200 events in wf 136.876 made in 1060pre3
ecalEnergy can apparently be negative and the fraction may be ill defined due to truncation.
This seems to be a rather common property of charged hadrons (not sure about other charged).
The fractions will be truncated at +/-1.28.
Please check/consider the cases where it may matter.

Reply:
See the other comment.